docs/features/delayed_creation/README.md

```diff
@@ -0,0 +1,130 @@
+// Create editor.
+var editor = CKEDITOR.replace( editorTargetElement );
+```
+
+Please notice, that we try to create editor on provided reference to the element. It is valid, but the element is no longer present in the DOM. And since editor initialization traverses the DOM and extracts information from the parent document of provided element it is not possible when target element is detached (it does not have parent in such cases). Such initialization attempt will result with an error.
+
+In the previous sample, editor could be initialized also by providing id to a editor element:
+
+```js
+var editor = CKEDITOR.replace( 'editor' );
+```
+
+However, this time the editor won't be created and the [editor-incorrect-element error](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_errors.html#editor-incorrect-element) error will be thrown. In this case, the code is looking for an element with given `id` in the document - which simply is not there, because it was detached earlier.
+
+## Practical example
+
+Lets assume we have:
+- a target element and it is not visible for the user (is detached),
+- button, which makes the element visible (attaches it to the DOM),
+- the editor creation was called with `detachIfDelay` option enabled.
+
+And then the user clicks the button, so the element becomes visible again (is attached to DOM). The pros and cons of using each approach are listed below.
+
+### Interval approach
+
+- Easy to use, doesn't require additional configuration.
+- The script will run until the button is clicked for the first time. In the worst case, the button may not be clicked at all, but the entire page is affected by the background script.
+- The last interval check may happen milliseconds before user clicks the button. The default timeout is small enough, that there will be no flickering effect, but still there will be slight delay before editor creation starts.
+
+### Callback approach
+
+- Requires additional configuration and good understanding when editor should be initialized.
+- Creation function can be used at any time which gives grater flexibility..
+- Because of invocation on demand, we need to wait only for creation time. No additional delays or unnecessary checks. However, additional logic needs to be implemented and binded to every place that could show the editor.
+
+## Getting editor reference
+
+There might be cases when you create editor and assign it to a variable. It is a common example in the entire documentation:
+
+```js
+var editor = CKEDITOR.replace( targetElement );
+```
+
+With this feature enabled and detached target element the variable value will not contain editor instance  (regardless of interval or callback method).
+
+```js
+var editor = CKEDITOR.replace( targetElement, {
+	delayIfDetached: true
+} );
+
+console.log( editor ); // -> null
+```
+
+You can get an instance in different ways, depending on used options:
+
+### Interval approach
+```js
+var editor = CKEDITOR.instances[ 'editorName' ];
+```
```

Comment:
I would mention here that getting editor ref should be done on `instanceReady` event. Because when this code is called before editor is created it will fail AFAIR.

Reply:
Well, you can grab an instance like that, but need to wait for `instanceReady` anyway to use API. Here, I just want to point out that `the old` way of creating an editor may return `null` whereas the object might be returned in the old code.

But, yes - I will mention that :)